src/azure-cli/HISTORY.rst

```diff
@@ -3,6 +3,119 @@
 Release History
 ===============
 
+2.4.0
+++++++
+
+**ACR**
+
+* `az acr run --cmd`: disable working directory override (#12877)
+* Dedicated data endpoint (#12967)
+
+**AKS**
+
+* az aks list -o table should show privateFqdn as fqdn for private clusters (#12784)
+* Add --uptime-sla (#12772)
+* Update containerservice package (#12964)
+* Add node public IP support in CLI extension. (#13015)
+* Fixing typo in the help command (#13055)
+
+**AppConfig**
+
+* Resolve key vault reference for kv list and export commands (#12893)
+
+**Appconfig**
+
+* Bug fix for list key values (#12926)
+
+**AppService**
+
+* `az functionapp create`: Changed the way linuxFxVersion was being set for dotnet linux function apps. This should fix a bug that was preventing dotnet linux consumption apps from being created. (#12817)
+* ([BREAKING CHANGE]) (az webapp create:) fix to keep existing AppSettings with az webapp create. (#12865)
+* ([BREAKING CHANGE]) (az webapp up:) fix to create RG for az webapp up command when using -g flag. (#12865)
+* ([BREAKING CHANGE]) (az webapp config:) fix to show values for non-JSON output with az webapp config connection-string list (#12865)
+
+**ARM**
+
+* `az deployment create/validate`: Add parameter `--no-prompt` to support skipping the prompt of missing parameters for ARM template (#11972)
+* `az deployment group/mg/sub/tenant validate`: Support comments in deployment parameter file (#12389)
+* `az deployment`: Remove `is_preview` for parameter `--handle-extended-json-format` (#12943)
+* `az deployment group/mg/sub/tenant cancel`: Support cancel deployment for ARM template (#12252)
+* `az deployment group/mg/sub/tenant validate`: Improve the error message when deployment verification fails (#12241)
+* `az deployment-scripts`: Add new commands for DeploymentScripts (#12928)
+* `az resource tag`: Add parameter `--is-incremental` to support adding tags to resource incrementally (#12736)
+
+**ARO**
+
+* `az aro`:  Add Azure RedHat OpenShift V4 aro command module (#12793)
+
+**Batch**
+
+* Update Batch API (#12813)
+
+**Compute**
+
+* `az sig image-version create`: Add storage account type Premium_LRS (#12919)
+* `az vmss update`: Fix terminate notification update issue (#12948)
+* `az vm/vmss create`: Add support for specialized image version (#12997)
+* SIG API Version 2019-12-01. (#12899)
+* `az sig image-version create`: Add --target-region-encryption. (#12899)
```

Comment:
append:
* Fix tests fail when running in serial due to keyvault name is duplicated in global in-momery cache